### PR TITLE
Removed limit in sparql and elastic when not given. Removed unnecessary arguments

### PR DESCRIPTION
**Pull request by crisely09**
This is a little effort to address two issues:
https://github.com/BlueBrain/nexus-forge/issues/241 and https://github.com/BlueBrain/nexus-forge/issues/262
Basically by not rewriting the offset and limit when they are given as `None`.

### CODE REVIEW

File: kgforge/core/archetypes/store.py [hunk 1]
```diff
@@ -369,29 +369,32 @@ def sparql(
...
         )
-        qr = _replace_in_sparql(qr, "LIMIT", limit, 100, r" LIMIT \d+")
-        qr = _replace_in_sparql(qr, "OFFSET", offset, 0, r" OFFSET \d+")
+        use_defaults = params.get("use_defaults", True)
+        if rewrite and use_defaults:
+            qr = _replace_in_sparql(qr, "OFFSET", offset, 0, r" OFFSET \d+")
+            qr = _replace_in_sparql(qr, "LIMIT", limit, 100, r" LIMIT \d+")
         if debug:
             self._debug_query(qr)
         return self._sparql(qr, limit, offset, **params)
```
Comment (ssssarah) [inline kgforge/core/archetypes/store.py:378]: sparql calls _sparql with 4 parameters when you've removed the last 2

File: kgforge/core/archetypes/store.py [hunk 2]
```diff
@@ -369,29 +369,32 @@ def sparql(
...
+        self, query: str, debug: bool, limit: int = None, offset: int = None
     ) -> List[Resource]:
         query_dict = json.loads(query)
-        query_dict["size"] = limit if limit else query_dict.get("size", 100)
+        if limit:
+            query_dict["size"] = limit
         query_dict["from"] = offset if offset else query_dict.get("from", 0)
         if debug:
             self._debug_query(query_dict)
         return self._elastic(json.dumps(query_dict), limit, offset)
```
Comment (ssssarah) [inline kgforge/core/archetypes/store.py:395]: Same here

**Review by ssssarah — CHANGES_REQUESTED** [review]

File: kgforge/core/archetypes/store.py (hunk 2 above)
Comment (crisely09) [reply kgforge/core/archetypes/store.py:395]: True, weird there are no tests failing!! maybe there are no tests for this.

**Review by ssssarah — DISMISSED** [review]

**Comment by crisely09** [comment]
I just added a couple of examples of the use-cases for those parameters to the getting-started notebooks.

**Comment by crisely09** [comment]
@MFSY I put back the elasticsearch default limit, but used a different parameter to `add_context` to a sparql query. I believe this can leave the door open to adding other possible parameters replacement when `rewrite` is set to True.

**Comment by crisely09** [comment]
@ssssarah  could you take another look, please?

File: kgforge/core/archetypes/store.py [hunk 3]
```diff
@@ -378,44 +378,49 @@ def search(
...
         rewrite = params.get("rewrite", True)
         qr = (
             rewrite_sparql(query, self.model_context, self.service.metadata_context)
             if self.model_context is not None and rewrite
             else query
         )
-        qr = _replace_in_sparql(qr, "LIMIT", limit, 100, r" LIMIT \d+")
-        qr = _replace_in_sparql(qr, "OFFSET", offset, 0, r" OFFSET \d+")
+        if limit:
+            qr = _replace_in_sparql(qr, "OFFSET", offset, 0, r" OFFSET \d+")
```
Comment (ssssarah) [inline kgforge/core/archetypes/store.py:398]: You swapped the conditions and the corresponding replacing statements

Comment (ssssarah) [reply kgforge/core/archetypes/store.py:398]: Also this is more of a format thing than a functionality thing, but 100 and 0 are hardcoded in many locations in these files, maybe having a global variable like SPARQL_DEFAULT_LIMIT & ELASTIC_DEFAULT_LIMIT + same for offset (or just DEFAULT_LIMIT and DEFAULT_OFFSET if it's known that both will always have the same defaults) could be better?

File: kgforge/core/archetypes/store.py (hunk 1 above)
Comment (ssssarah) [reply kgforge/core/archetypes/store.py]: This is still a problem here

File: kgforge/core/archetypes/store.py (hunk 3 above)
Comment (crisely09) [reply kgforge/core/archetypes/store.py:398]: yeah, DEFAULT_LIMIT and DEFAULT_OFFSET could work 👍🏽

Comment (ssssarah) [reply kgforge/core/archetypes/store.py]: If a limit is hardcoded, but a limit is specified in the function, the function value should override the hardcoded limit. This works when a number is specified, but if it is specified that the limit should be None, it keeps the hardcoded value. This feels unintentional. I'd think the limit should be removed altogether, no?

**Review by ssssarah — APPROVED** [review]

**Pull request merged**